### PR TITLE
[DM-29532] Code cleanup

### PR DESCRIPTION
**Pull request by cbanek**

### CODE REVIEW

File: dev-values.yaml [hunk 1]
```diff
@@ -91,22 +91,21 @@ jupyterhub:
...
+      ram: 1536M
+    - name: Small
+      cpu: 1
+      ram: 3072M
+    - name: Medium
+      cpu: 2
+      ram: 6144M
+    - name: Large
+      cpu: 4
+      ram: 12288M
```
Comment (cbanek) [inline dev-values.yaml:108]: pulling this in one level, we'll need to do this in the chart as well.

**Review by athornton — COMMENTED** [review]
Only two things that concern me, and maybe they are nothing to worry about:

if we're making auto_repo_urls an option, then maybe we want to do the same trick we did with image info to turn it into a class that returns a string (comma separated URLs) when used in the environment context but is presented as a list in whatever yaml it's defined?  The basic problem is the same: at point-of-use it's a string, but it's a string that serializes a more complicated data structure.

The other thing is that I don't really get what's going on with packed_string in the imageinfo class, but as long as the class is still a frozen dataclass, and the two classmethod factories give back an appropriately-populated ImageInfo, it's fine.  And there are tests for that, so I presume it's still doing the right thing.

**Review by rra — APPROVED** [review]
Some nits but it all looks good to me.

File: src/nublado2/nublado_config.py [hunk 2]
```diff
@@ -2,26 +2,69 @@
...
+        with open("/etc/keys/signing_key.pem", "r") as f:
+            return f.read()
 
-        for s in sizes:
-            if s["name"] == name:
-                return (float(s["cpu"]), s["ram"])
+    @property
+    def sizes(self) -> Dict[str, LabSize]:
+        """Retrieve the sizes a lab can spawn as."""
+        return self._sizes
```
[the remark this answers is not in the document]
Comment (cbanek) [reply src/nublado2/nublado_config.py:65]: Fixed thanks!

File: src/nublado2/nublado_config.py [hunk 3]
```diff
@@ -2,26 +2,69 @@
...
 
-        raise ValueError(f"Size {name} not found")
+    @property
+    def user_resources(self) -> List[Any]:
+        """Retrieve the jinja template for creating lab resources."""
+        return self._config.get("user_resources", [])
```
[the remark this answers is not in the document]
Comment (cbanek) [reply src/nublado2/nublado_config.py:70]: Excellent, thanks

File: src/nublado2/nublado_config.py [hunk 4]
```diff
@@ -2,26 +2,69 @@
...
+            for s in [
+                LabSize(float(s["cpu"]), s["name"], s["ram"])
+                for s in self._config["options_form"]["sizes"]
+            ]
+        }
+
+    @property
+    def auto_repo_urls(self) -> str:
+        """A list of URLs of repositories for the lab to clone on start."""
+        return self._config.get("auto_repo_urls", "")
```
[the remark this answers is not in the document]
Comment (cbanek) [reply src/nublado2/nublado_config.py]: Okay tore this out for now.

**Pull request merged**